server/api/src/rsModColl.cpp

```diff
@@ -106,8 +111,11 @@ auto rsModColl(rsComm_t* rsComm, collInp_t* modCollInp) -> int
         // server-to-server connection. Many operations in iRODS use this API to update the collection mtime after a
         // data object modification or creation and requires privileged status to modify the collection in the event
         // that the user does not have permissions on the parent collection. A client connection is made here using the
-        // service account rodsadmin credentials to modify the collection and then is immediately disconnected.
-        if (irods::is_privileged_client(*rsComm)) {
+        // service account rodsadmin credentials to modify the collection and then is immediately disconnected. This is
+        // only done for connections with elevated privileges that are not rodsadmins.
+        if (irods::is_privileged_client(*rsComm) &&
+            ua::user_type::rodsadmin != *ua::server::type(*rsComm, ua::user{rsComm->clientUser.userName}))
```

Comment:
Does this cover remote users?

Reply:
See comment in `rsModAccessControl`.

I'll add a test similar to `test_remove_data_object_in_collection_with_read_permissions__issue_6428` to the federation test suite.

Reply:
Done